### PR TITLE
Add labels to k8s jobs for the PodTemplateTaskAdapter

### PR DESCRIPTION
**Pull request by georgew5656**
We add taskId/taskType/groupid/dataSource annotations to jobs launched with the K8s task runner for information but its nice to have these in labels as well to e.g. filter jobs by data source in kubectl or in something like kubecost. (These tools generally support grouping by labels more than grouping by annotations).

These key/value pairs are all pretty small so I don't think this will cause any issues with the size of the spec.

### Description
Add the same key/value pairs that we add to annotations to labels as well (excluding the actual task blob)
I added a druid.io prefix for the labels to follow k8s label naming conventions

#### Release note
- Add additional labels to K8s Task Runner jobs

##### Key changed/added classes in this PR
Update the PodTemplateTaskAdapter to include some additional helpful labels

This PR has:

- [X ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [X ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by rmn7 — APPROVED** [review]

File: extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidK8sConstants.java [hunk 1]
```diff
@@ -36,7 +36,8 @@
   public static final String TASK_ID_ENV = "TASK_ID";
   public static final String JAVA_OPTS = "JAVA_OPTS";
   public static final String DRUID_HOST_ENV = "druid_host";
-  public static final String DRUID_HOSTNAME_ENV = "HOSTNAME";
+  public static final String DRUID_HOSTNAME_ENV = "HOSTNAME"
```
Comment (imply-elliott) [inline extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidK8sConstants.java:39]: Did you mean to remove the semicolon here?

Comment (georgew5656) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidK8sConstants.java:39]: oops

Comment (georgew5656) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidK8sConstants.java]: should be good now

**Review by dclim — APPROVED** [review]

File: extensions-contrib/kubernetes-overlord-extensions/src/test/resources/expectedNoopJob.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ metadata:
   name: "id"
   labels:
     druid.k8s.peons: "true"
+    druid.io/task.id: "id"
+    druid.io/task.type: "noop"
+    druid.io/task.group.id: "id"
+    druid.io/task.datasource: "datasource"
```
Comment (abhishekagarwal87) [inline extensions-contrib/kubernetes-overlord-extensions/src/test/resources/expectedNoopJob.yaml:10]: can we use the same convention as `druid.k8s.peons`? 
so `druid.io/task.id` becomes `druid.task.id` - the latter is also easier to type.

**Review by abhishekagarwal87 — APPROVED** [review]

**Review by asdf2014 — APPROVED** [review]

**Pull request merged**